File: src/Challenge235EASYRuthAaronPairs.java
import java.util.Scanner;

/**
 * Created by Andrew on 12/8/15.
 */
public class Challenge235EASYRuthAaronPairs {
    public static void main(String args[]){
        SBUtil util = new SBUtil();
        Scanner stdIn = new Scanner(System.in);

        while(true){
            util.printl("Enter a tuple(type 'quit' to quit): ");
            String s = stdIn.nextLine();
            if(s.contains("quit"))
                break;
            String temp = s.replace(")", "").replace("(", "").replace(" ", "");
            int n1 = Integer.parseInt(temp.split(",")[0]);
            int n2 = Integer.parseInt(temp.split(",")[1]);
            boolean valid = false;
            if(factorSum(n1) == factorSum(n2))
                valid = true;
            if(valid)
                util.printl(s + " VALID");
            else
                util.printl(s + " NOT VALID");
        }
        util.goodbye();
    }

    public static int factorSum(int n){
        int sum = 0;
        for(int i = 2; i <= n; ++i){
            if(n % i == 0){
                sum += i;
                while(n % i == 0)
                    n = n/i;
            }
        }
        return sum;
    }
}

File: Python/Challenge235EASYRuthAaronPairs.py
# Created by Andrew Budziszek on 12/25/15
# Challenge #235[EASY]: Ruth-Aaron Pairs
# https://www.reddit.com/r/dailyprogrammer/comments/3nkanm/20151005_challenge_235_easy_ruthaaron_pairs/


def factor_sum(n):
    return_sum = 0
    n = int(n)
    for i in range(2, n):
        if n % i == 0:
            return_sum += i
            while n % i == 0:
                n /= i
    return return_sum


while True:
    s = raw_input('Enter a tuple(\'exit\' to quit): ')
    if s == 'exit':
        break
    s = s.replace(')', '').replace('(', '').replace(' ', '')
    n1 = s.split(',')[0]
    n2 = s.split(',')[1]
    valid = False
    if factor_sum(n1) == factor_sum(n2):
        valid = True
    if valid:
        print s + " is VALID"
    else:
        print s + " is NOT VALID"
print "Goodbye!"
